src/hotspot/share/cds/archiveBuilder.cpp

```diff
@@ -312,7 +312,7 @@ void ArchiveBuilder::sort_klasses() {
 }
 
 address ArchiveBuilder::reserve_buffer() {
-  size_t buffer_size = LP64_ONLY(CompressedClassSpaceSize) NOT_LP64(256 * M);
+  size_t buffer_size = LP64_ONLY(CompressedClassSpaceSize) NOT_LP64(256 * M) + ReservedCodeCacheSize;
```

Comment:
Please add comment about `ReservedCodeCacheSize`. Why not `CachedCodeMaxSize`?

Reply:
> Why not CachedCodeMaxSize?

Yeah, I think that makes sense. Since the aot code buffer is based on CachedCodeMaxSize, we can use this for sizing cached code region as well.